tests/ops/op_math/test_controlled.py

```diff
@@ -39,6 +39,36 @@
 # pylint: disable=pointless-statement
 # pylint: disable=expression-not-assigned
 
+
+def equal_if_decomposed(lhs, rhs, max_iterations=1000):
+    """
+    Test that 2 operators or operator sequences are the same if fully decomposed.
```

Comment:
What does fully decomposed even mean?

Reply:
The intention is to decompose until we reach unambiguous basic operations, which I believe (correct me if I'm wrong) means the single qubit gates plus CNOT. It might allow Toffoli too, which seems to be a non-issue for now. The more complex operations often have multiple equivalent forms using different classes which `qml.equal` won't recognize as equal

Reply:
`equal_if_decomposed` has been removed, now we check exact equality with `equal_list`